### PR TITLE
fix gitRemote null reference

### PR DESCRIPTION
**Pull request by superyyrrzz**
#1886

### CODE REVIEW

**Comment by hellosnow** [comment]
:shipit:

File: src/docfx.website.themes/common/common.js [hunk 1]
```diff
@@ -150,29 +150,26 @@ function getRemoteUrl(remote, startLine, gitContribute, gitUrlPattern) {
...
-    }
+    return result;
 
-    return {
-        apiSpecFolder: gitContribute.apiSpecFolder || defaultApiSpecFolder,
-        repo: gitContribute.repo || gitRemote.repo,
-        branch: gitContribute.branch || gitRemote.branch,
-        path: gitRemote.path
+    function mergeKey(source, sourceFallback, dest, key) {
+        if (source && source[key]) {
```
Comment (vicancy) [inline src/docfx.website.themes/common/common.js]: as a general function, should we provide a more strict check to check only `undefined` ? For example, `false` or string.Empty all returns false currently

Comment (superyyrrzz) [inline src/docfx.website.themes/common/common.js]: agree. fixed it.

File: src/docfx.website.themes/common/common.js [hunk 2]
```diff
@@ -150,29 +150,26 @@ function getRemoteUrl(remote, startLine, gitContribute, gitUrlPattern) {
...
-    }
+    return result;
 
-    return {
-        apiSpecFolder: gitContribute.apiSpecFolder || defaultApiSpecFolder,
-        repo: gitContribute.repo || gitRemote.repo,
-        branch: gitContribute.branch || gitRemote.branch,
-        path: gitRemote.path
+    function mergeKey(source, sourceFallback, dest, key) {
+        if (source && key in source) {
```
Comment (vicancy) [inline src/docfx.website.themes/common/common.js]: use `hasOwnProperty`? `in` returns true for `prototype` properties.

Comment (superyyrrzz) [inline src/docfx.website.themes/common/common.js]: resolved

**Comment by 928PJY** [comment]
:shipit:

**Pull request merged**